debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbStreamingChangeEventSource.java

```diff
@@ -318,6 +291,94 @@ protected MongoDbOffsetContext initializeOffsets(MongoDbConnectorConfig connecto
     }
 
+    /**
+     * A factory to produce a MongoDB change stream pipeline expression.
+     */
+    static class ChangeStreamPipelineFactory {
+
+        private final MongoPreferredNode mongo;
+        private final ReplicaSetOffsetContext rsOffsetContext;
+        private final MongoDbConnectorConfig connectorConfig;
+
+        ChangeStreamPipelineFactory(MongoPreferredNode mongo, ReplicaSetOffsetContext rsOffsetContext, MongoDbConnectorConfig connectorConfig) {
+            this.mongo = mongo;
+            this.rsOffsetContext = rsOffsetContext;
+            this.connectorConfig = connectorConfig;
+        }
+
+        List<Bson> create() {
+            // Resolve the leaf filters
+            var filters = Stream
+                    .of(
+                            createCollectionFilter(mongo),
+                            createOperationTypeFilter(connectorConfig),
+                            createClusterTimeFilter(rsOffsetContext))
+                    .flatMap(Optional::stream)
+                    .collect(toList());
+
+            // Combine
+            var andFilter = Filters.and(filters);
+            var matchFilter = Aggregates.match(andFilter);
+            var pipeline = Collections.singletonList(matchFilter);
+            LOGGER.debug("Change stream pipeline: {}", pipeline);
+
+            return pipeline;
+        }
+
+        private static Optional<Bson> createCollectionFilter(MongoPreferredNode mongo) {
+            // Get collection filters on change stream pipeline. Note that `MongoPreferredNode#collections`
+            // performs the configured filtering by regex
+            // See https://www.mongodb.com/docs/manual/reference/change-events/create/#mongodb-data-create
+            var namespaces = mongo.collections().stream()
+                    .map(c -> new Document()
+                            .append("db", c.dbName())
+                            .append("coll", c.name()))
+                    .collect(toList());
+
+            return Optional.of(Filters.in("ns", namespaces));
+        }
+
+        private static Optional<Bson> createOperationTypeFilter(MongoDbConnectorConfig connectorConfig) {
+            // Per https://debezium.io/documentation/reference/stable/connectors/mongodb.html#mongodb-property-skipped-operations
+            // > The support operations include:
+            // > - 'c' for inserts/create, '
+            // > - 'u' for updates,
+            // > - 'd' for deletes,
+            // > - 't' for truncates, and
+            // > - 'none' to not skip any operations.
+            // > By default, 'truncate' operations are skipped (not emitted by this connector).
+            // However, 'truncate' is not supported since it doesn't exist as a MongoDB change type. And support for 'none'
+            // seems to effectively imply 'c', 'u', 'd'
+            // TODO: What about the rest of these when 'none' is selected? https://www.mongodb.com/docs/manual/reference/change-events/
```

Comment:
Concern cited below with respect to docs. 

As an aside, to make the code more readable / explicit, I was thinking of defining the allowed operations explicitly in a set, then removing based on the config.

Reply:
Sounds good

Reply:
Updated.